test/e2e/logforwarding/fluent/fluent_test.go

```diff
@@ -0,0 +1,154 @@
+package fluent_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	loggingv1 "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test"
+	"github.com/openshift/cluster-logging-operator/test/client"
+	"github.com/openshift/cluster-logging-operator/test/helpers/cmd"
+	"github.com/openshift/cluster-logging-operator/test/helpers/fluentd"
+	. "github.com/openshift/cluster-logging-operator/test/matchers"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const message = "My life is my message"
+
+var (
+	// Create/delete clusterlogging instance around entire suite.
+	cl = runtime.NewClusterLogging()
+	_  = BeforeSuite(func() { ExpectOK(client.Get().Recreate(cl)) })
+	_  = AfterSuite(func() { _ = client.Get().Remove(cl) })
+)
+
+// Extend client.Test for tests using logGenerator and receiver.
+type Test struct {
+	*client.Test
+	receiver     *fluentd.Receiver
+	logGenerator *corev1.Pod
+	group        test.FailGroup // Run cluster operations in parallel
+}
+
+func (t *Test) Reader(name string) *cmd.Reader {
+	return t.receiver.Sources[name].TailReader()
+}
+
+func NewTest(message string) *Test {
+	ct := client.NewTest()
+	t := &Test{
+		Test:         ct,
+		receiver:     fluentd.NewReceiver(ct.NS.Name, "receiver"),
+		logGenerator: runtime.NewLogGenerator(ct.NS.Name, "log-generator", 10000, 0, message),
+	}
+	t.group.Go(func() { ExpectOK(t.Create(t.logGenerator)) })
+	return t
+}
+
+func (t *Test) Close() {
+	defer t.Test.Close()
+	t.group.Wait()
+}
+
+var _ = Describe("[ClusterLogForwarder]", func() {
+	var t *Test
+	BeforeEach(func() { t = NewTest(message) })
+	AfterEach(func() { t.Close() })
+
+	Context("with app/infra/audit receiver", func() {
+		BeforeEach(func() {
+			t.receiver.AddSource("application", "forward", 24224)
+			t.receiver.AddSource("infrastructure", "forward", 24225)
+			t.receiver.AddSource("audit", "forward", 24226)
+			t.group.Go(func() { ExpectOK(t.receiver.Create(t.Client)) })
+		})
+
+		It("forwards application logs only", func() {
+			clf := runtime.NewClusterLogForwarder()
+			addPipeline(clf, t.receiver.Sources["application"])
+			t.group.Go(func() { ExpectOK(t.Recreate(clf)) })
+			// Use ctx to let empty checks run until we get the expected lines.
+			ctx, cancel := context.WithCancel(context.Background())
+			t.group.Go(func() {
+				defer cancel() // Cancel goroutines verifying empty when we get our logs.
+				r := t.Reader("application")
```

Comment:
Maybe more verbose of what I'm getting?   "ReaderForSourceType(" ??

Reply:
Done.